settings/acsf.local.settings.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+use Acquia\Blt\Robo\Config\ConfigInitializer;
+use Symfony\Component\Console\Input\ArgvInput;
+
+$input = new ArgvInput($_SERVER['argv']);
+$config_initializer = new ConfigInitializer($repo_root, $input);
+$config = $config_initializer->initialize();
+
+$name = substr($_SERVER['HTTP_HOST'],0, strpos($_SERVER['HTTP_HOST'],'.local'));
+$acsf_sites = $config->get('acsf.sites');
+if (in_array($name, $acsf_sites)) {
+  /**
+   * Database configuration.
+   */
+  $databases = array(
+    'default' =>
+      array(
+        'default' =>
+          array(
+            'database' => "drupal_{$name}",
+            'username' => 'drupal',
+            'password' => 'drupal',
+            'host' => 'localhost',
+            'port' => '3306',
+            'namespace' => 'Drupal\\Core\\Database\\Driver\\mysql',
+            'driver' => 'mysql',
+            'prefix' => '',
+          ),
+      ),
+  );
+}
```

Comment:
It'd be good to also add dynamic files and config folder paths here:

   $config_directories['sync'] = $dir . '/config/' . $name . '/local';
+
+    /**
+     * Private file path.
+     */
+    $settings['file_private_path'] = $dir . "/files-private/${name}";
+
+    /**
+     * Public file path.
+     */
+    // Set public filesystem per local development site.
+    $settings['file_public_path'] = "sites/default/files/${name}";
+  }

Reply:
I've updated the pull request to set the file paths dynamically. However, I'm not convinced that the configuration directory structure that you suggested makes sense as a default.

My thinking is that most applications will want to use the following configuration directory structure:
- config
    - default:  default configuration, shared by all sites
    - envs:
        - local:  local environment configuration, shared by all sites
    - site1:  configuration for docroot/sites/site1
    - site2: configuration for docroot/sites/site2

It would still be possible for each site to have its own specialized local configuration, but that seems overly granular as a default. Nothing prohibits you from taking that approach by setting custom configuration directories in your applications settings.php file. 

At present, BLT will automatically set the status of the site1 configuration split to `true` when the normal multisite approach is being used (I.E., non ACSF). do we want these site-specific splits to be enabled automatically in non-local environments?  I would think that having a bit of custom logic in your settings.php file to activate splits for site factory sites would be the safest approach here.

Reply:
I would err on the side of enabling the splits on ACSF or providing a task that generates the needed factory hook to encourage / enforce config management best practices which is typically a pain point on ACSF and difficult to debug locally.  With improved multisite and config split support in BLT it will be much easier to maintain environment, site, and profile-specific layers of configuration on ACSF so I would encourage enabling the splits from the onset and making the decision to disable them a conscious one by the developer.

Reply:
@lcatlett  How would you suggest determining which split should be enabled on the site factory environment? In the local environment, we are using the host name. Do you think that that is sufficient?